packages/interactivity-router/src/index.js

```diff
@@ -229,17 +305,26 @@ export const { state, actions } = store( 'core/router', {
 		 * The function normalizes the URL and stores internally the fetch
 		 * promise, to avoid triggering a second fetch for an ongoing request.
 		 *
-		 * @param {string}  url             The page URL.
-		 * @param {Object}  [options]       Options object.
-		 * @param {boolean} [options.force] Force fetching the URL again.
-		 * @param {string}  [options.html]  HTML string to be used instead of
-		 *                                  fetching the requested URL.
+		 * @param {string|Event} eventOrUrl      The page href or the event handler in case it is used directly in a directive.
+		 * @param {Object}       [options]       Options object.
+		 * @param {boolean}      [options.force] Force fetching the URL again.
+		 * @param {string}       [options.html]  HTML string to be used instead of fetching the requested URL.
 		 */
-		prefetch( url, options = {} ) {
+		prefetch( eventOrUrl, options = {} ) {
```

Comment:
Same here, can we separate this into two actions, one expecting `event`, the other expecting `url`?